docs/docs/wikitext103.md

```diff
@@ -0,0 +1,205 @@
+# WikiText-103
+
+![Banner](img/banner.png)
+[//]: # (TODO Change Me) 
+
+You can download WikiText-103 dataset [here](https://blog.einstein.ai/the-wikitext-long-term-dependency-language-modeling-dataset/), there is no official leaderboard.
+
+## TLDR;
+
+Here is a minimal working example that let you evaluate `transformer-xl` from `huggingface/transformers`.
+
+```python
+import torch
+from sotabencheval.langauge_modelling import WikiText103Evaluator
+
+model = torch.hub.load('huggingface/transformers', 'modelWithLMHead', 'transfo-xl-wt103').to("cuda")
+tokenizer = torch.hub.load('huggingface/transformers', 'tokenizer', 'transfo-xl-wt103').to("cuda")
+
+with WikiText103Evaluator.dataset.get_path('local/dir/wikitext-103').open() as f:
+    test_data = torch.tensor(tokenizer.encode(f.read())).to("cuda").eval()
+
+def log_probs_generator(model, test_data, seq_len=128):
+    with torch.no_grad():
+        X, Y, mems = test_data[None, :-1], test_data[None, 1:], None
+        for s in range(0, X.shape[-1], seq_len):
+            log_probs, mems, *_ = model(input_ids=X[s:s+seq_len], mems=mems)
+            yield log_probs, y[s:s+seq_len]
+
+WikiText103Evaluator(
+    model_name="Transformer-XL Large", 
+    paper_arxiv_id="1901.02860",
+    paper_pwc_id="transformer-xl-attentive-language-models" 
+).eval(log_probs_generator(model, test_data)).print_results()
```

Comment:
confused. also there's no evaluator.save? Ideally we would want consistent API between different benchmarks.

Reply:
This just an example of a wrapper that could simplify things as you don't have to remember about `save`, `check_cache`, `reset`, `reset_timer` and exception handling.

Reply:
But as said before, I will update docs to have the same api.

Reply:
yep makes sense, but let's keep things consistent for now and improve things later. One way to think about this is that we'll get feedback on the API upon release and how people are using it. Then we can figure out what to keep, and what we don't need. Then once we have this high quality feedback, we can figure out what to wrap and simplify (similar to your suggestions).